directory.cpp

```diff
@@ -58,42 +77,86 @@ void directory::rm(string todelete)
 
 }
 void directory::segedrmrf(){
-    for(auto it=subdirectories.begin();it!=subdirectories.end();){
-        if((*it)->hasDirs()){
-            (*it)->segedrmrf();
-        }
-        else if(!(*it)->hasDirs()){
-            it=this->subdirectories.erase(it);
+    for(auto it=subdirectories.begin();it!=subdirectories.end();)
+    {
+        directory* temp=dynamic_cast<directory*>(*it);
+        if(temp!=nullptr){
+            if(temp->hasDirs()){
```

Comment:
ez az if minek?

Reply:
Na, @patrikkocsis ha az rmrf torli azt is, amire meg lett hivva, akkor nyugodtan meg lehet hivni rekurzivan az ures konyvtarakra is. Ha meg nem, akkor meg a nemures konyvtarakat kesobb sem torlod ki, mert az else if-be mar nem mesz be. Az Else agba meg sosem mesz bele.